src/printer.js

```diff
@@ -3024,8 +3029,21 @@ function printJSXElement(path, options, print) {
   let forcedBreak = willBreak(openingLines);
 
   const jsxWhitespace = options.singleQuote
-    ? ifBreak("{' '}", " ")
-    : ifBreak('{" "}', " ");
+    ? ifBreak(concat([softline, "{' '}", softline]), " ")
```

Comment:
the mutations involved later on with this seem pretty bad... but I think you may be on the right track in general. How about something like this: 

1) change this to: `const jsxWhitespace = options.singleQuote ? "{' '}" : '{" "}';`
2) where `jsxWhitespace` is used above, change its usage to `ifBreak(concat([jsxWhitespace, softline]), " ")` (with the appropriate usage of `softline`, typically as indicated in the former code). 

Thoughts?